com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/configuration/QuartzConfiguration.java

```diff
@@ -73,4 +76,27 @@ public SchedulerFactoryBean schedulerFactoryBean(DataSource dataSource, DataSour
         return schedulerFactoryBean;
     }
 
+    @Inject
+    @Bean
+    @Profile("hana")
+    public SchedulerFactoryBean schedulerFactoryBeanHana(DataSource dataSource, DataSourceTransactionManager transactionManager,
+                                                         AutowiringSpringBeanJobFactory quartzJobFactory, JobDetail jobDetail,
+                                                         Trigger trigger) {
+        SchedulerFactoryBean schedulerFactoryBean = new SchedulerFactoryBean();
+        schedulerFactoryBean.setDataSource(dataSource);
+        schedulerFactoryBean.setTransactionManager(transactionManager);
+        schedulerFactoryBean.setJobFactory(quartzJobFactory);
+        schedulerFactoryBean.setOverwriteExistingJobs(true);
+        schedulerFactoryBean.setSchedulerName("cleanupScheduler");
+        schedulerFactoryBean.setApplicationContextSchedulerContextKey("applicationContext");
+        JobDetail[] jobDetails = Arrays.asList(jobDetail)
+                                       .toArray(new JobDetail[0]);
+        schedulerFactoryBean.setJobDetails(jobDetails);
+        Trigger[] triggers = Arrays.asList(trigger)
+                                   .toArray(new Trigger[0]);
+        schedulerFactoryBean.setTriggers(triggers);
```

Comment:
Can you somehow reuse this code in order to avoid double maintenance?

Reply:
Done, 10x